File: server/internal/model/plan.go
package model

import (
	"time"

	"github.com/google/uuid"

	"wsTrack/server/internal/enum"
)

type Plan struct {
	ID          uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	UserID      uuid.UUID `gorm:"type:uuid;index;not null"`
	User        User      `gorm:"foreignKey:UserID"`
	Name        string    `gorm:"size:100;not null"`
	Description *string
	Goal        *enum.PlanGoal `gorm:"type:varchar(32)"`
	IsActive    bool           `gorm:"default:false"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
	Days        []PlanDay `gorm:"foreignKey:PlanID;constraint:OnDelete:CASCADE"`
}

File: mobile/src/models/Plan.ts
import { Model } from "@nozbe/watermelondb";
import type { Query } from "@nozbe/watermelondb";
import type { Associations } from "@nozbe/watermelondb/Model";
import { children, field } from "@nozbe/watermelondb/decorators";

import type { PlanGoal } from "../constants/enums";
import type { PlanDay } from "./PlanDay";

export class Plan extends Model {
  static table = "plans";

  static associations: Associations = {
    plan_days: { type: "has_many", foreignKey: "plan_id" },
  };

  @field("name") name!: string;
  @field("description") description!: string | null;
  @field("goal") goal!: PlanGoal | null;
  @field("is_active") isActive!: boolean;
  @field("is_archived") isArchived!: boolean;
  @field("created_at") createdAt!: number;
  @field("updated_at") updatedAt!: number;

  @children("plan_days") days!: Query<PlanDay>;
}
